daemon/libnetwork/types/types.go

```diff
@@ -56,6 +56,8 @@ type PortBinding struct {
 	HostIP      net.IP
 	HostPort    uint16
 	HostPortEnd uint16
+	Mapper      string
```

Comment:
Wait; this is the type used for the API response? 😬


Can we have a type defined somewhere in the API module? We started to consider anything inside the daemon and libnetwork internal (and at most an internal API); if these types are directly used for the API we risk making changes to the API without knowing

Reply:
> Wait; this is the type used for the API response? 😬

No, it's not directly part of the API response (i.e. the api module doesn't import it).

But it's the type used by the daemon to tell libnetwork drivers which port bindings should be created, and for drivers to report back which port bindings have been created.

So in order to pass the new Mapper and ExtraParams fields from the API request to libnetwork, I need to change that type too.